client/beefy/src/worker.rs

```diff
@@ -400,6 +408,11 @@ where
 		));
 
 		loop {
+			if self.sync_oracle.is_major_syncing() {
+				debug!(target: "beefy", "Skipping initialization due to sync.");
+				continue
+			}
+
```

Comment:
I feel like this log would be a bit too noisy, is there anyway we can make a function `wait_for_major_syncing` that returns a future that returns when major syncing ends?

```suggestion
			if self.sync_oracle.is_major_syncing() {
				debug!(target: "beefy", "Skipping initialization due to sync.");
				self.sync_oracle..wait_for_major_syncing().await;
			}

```